### DISCUSSION TITLE
June 2021 Priorities, Part 1

### DISCUSSION THREAD

**Discussion by jefftangx** [athensresearch/athens#1271] [Goals]
# [June 2021 Priorities, Part 2 Here](https://github.com/athensresearch/athens/discussions/1354)

Notes: https://roamresearch.com/#/app/athensresearch/page/wzoTtozBy

# Backend

<details>
<summary>P1: Self-Hosted Collaboration Using Single-Writer Approach | Status: In Progress | @neotyk </summary>

- Why: our top customer needs this and no one else has real-time collaboration yet (except for Roam which has really bad collaboration UX and still has data loss).
- Main PR outlining approach #1170 
- Getting support from Flexiana and Datahike

</details>

# Frontend

<details>
<summary>P2: Text Editor | Status: Help Wanted | @shanberg </summary>

- Why: lots of bugs occur from our text editor implementation which has two views: render (when not focused) and raw (when focused)
- See #1091, #566 

</details> 

<details>
<summary>P2: DB Switching | Status: In Progress | @shanberg @sid597 </summary>

- Why: improves UX for when we have collaborative knowledge graphs -- will want to be able to switch between local and remote graphs easily.
- #882 

</details> 

<details>
<summary>P2: Presence | Status: scoping | @shanberg @neotyk </summary>

- Why: collaboration UX is very important for collaborative graphs
- See ["Presence" search query](https://github.com/athensresearch/athens/issues?q=is%3Aissue+is%3Aopen+presence)
- [Notes on discussion between @shanberg and @neotyk](https://roamresearch.com/#/app/athensresearch/page/aEfAXnSHX)

</details> 

<details>
<summary>P2: Utilize a dropdown/popover library | Status: Ready to start | @shanberg @juniusfree </summary>

- Why: TODO #1316
- https://github.com/athensresearch/athens/issues/869

</details> 

# Community

<details>
<summary>P1: Head of Community | Status: Done | @joelhans </summary>

- Why: community is a core function/domain of Athens and we need a rock solid pillar that can build out this function and team. 
- Continue to refine [job description](https://www.notion.so/athensresearch/Athens-is-Hiring-Job-Descriptions-7f52ac86e1e1406f93624503e226dfe6#2a769f1161d043e2a8ccb90fe4828654)
- Learn from Heads of Community in other OSS and Crypto Startups/Projects
- @joelhans starting full-time June 18th

</details> 

<details>
<summary>P1: Finish the Handbook  | Status: In Progress | @joelhans </summary>

- Why: we need to document our guidelines for both contributors and employees in order to effectively and efficiently collaborate at scale
- @joelhans taking over

</details> 

<details>
<summary>P2: Scaling GitHub | Status: Help Wanted | @tangjeff0 </summary>

- Why: we need to master GitHub features to build Athens at scale
- [Research, notes, links on Roam](https://roamresearch.com/#/app/athensresearch/page/bto7svyxW)
- [The current problem, Roam notes](https://roamresearch.com/#/app/athensresearch/page/ezxVSVobZ)

</details> 

<details>
<summary>P1: Launch Website | Status: so close | @tangjeff0 @shanberg @mateicanavra </summary>

- Why: we need a 1) a central entrypoint to Athens that is not GitHub/Twitter/Discord/Handbook, 2) we need to begin marketing Athens to our ideal customer to gain adoption and monetization

</details> 

# Meta

<details>
<summary>P1: Onboard Core Team | Status: In Progress | @tangjeff0 @shanberg @mateicanavra @joelhans @neotyk </summary>

- Why: we are bringing on people to build out entire departments of an open-source startup. We are going 1->5 core leaders, which requires evolution in terms of communication and knowledge and task management infrastructure.
- create and document systems, processes, guidelines for work, meetings, etc.

</details> 

# [Retrospective on May](https://roamresearch.com/#/app/athensresearch/page/ZJR6yWIzx)

**Comment by juniusfree** [comment]
> P2: Utilize a dropdown/popover library | Status: Needs Research

@tangjeff0 Can I try to do this again?

**Reply by jefftangx** [reply]
@juniusfree requires research on which libraries we want to use. @shanberg is unsure about Material UI Dropdown component, I am okay with it. Can you evaluate the current options available to and share a doc?

**Reply by shanberg** [reply]
The Material UI dropdown works fine, and nothing's stopping us from using it for our context menus, dropdowns, tooltips, and dialogs.

I just want to evaluate it more carefully before we get so deep into it by accident that we can't afford to eject.

**Reply by juniusfree** [reply]
@shanberg What are the considerations before we choose a library?

**Reply by shanberg** [reply]
@juniusfree I just wanted to make sure I understood what we were getting with Material-UI, and what potential drawbacks might be over time because it felt like we'd just stumbled into the library without examining the options. But I'm happy with what I've seen, so I have no reservations.

**Reply by juniusfree** [reply]
@shanberg Alright. I can start with the conversion of the pop-ups if that's okay.
